internal/services/kusto/kusto_database_script_resource.go

```diff
@@ -51,28 +51,42 @@ func resourceKustoDatabaseScript() *pluginsdk.Resource {
 				ValidateFunc: validate.DatabaseID,
 			},
 
-			"url": {
+			"continue_on_errors_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+
+			"force_an_update_when_value_changed": {
 				Type:         pluginsdk.TypeString,
-				Required:     true,
+				Optional:     true,
+				Computed:     true,
 				ValidateFunc: validation.StringIsNotEmpty,
 			},
 
-			"sas_token": {
+			"url": {
 				Type:         pluginsdk.TypeString,
-				Required:     true,
+				Optional:     true,
+				ExactlyOneOf: []string{"url", "script_content"},
+				RequiredWith: []string{"sas_token"},
 				ValidateFunc: validation.StringIsNotEmpty,
 			},
 
-			"continue_on_errors_enabled": {
-				Type:     pluginsdk.TypeBool,
-				Optional: true,
-				Default:  false,
+			"sas_token": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				RequiredWith: []string{"url"},
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotEmpty,
 			},
 
-			"force_an_update_when_value_changed": {
+			"script_content": {
 				Type:         pluginsdk.TypeString,
+				ExactlyOneOf: []string{"url", "script_content"},
 				Optional:     true,
-				Computed:     true,
+				ForceNew:     true,
+				Sensitive:    true,
```

Comment:
Does this need to be sensitive?

Reply:
I added this according to this in swagger
`"scriptContent": {
          "description": "The script content. This property should be used when the script is provide inline and not through file in a SA. Must not be used together with scriptUrl and scriptUrlSasToken properties.",
          "type": "string",
          "x-ms-mutability": [
            "create"
          ],
          "x-ms-secret": true
        },`